allensdk/brain_observatory/behavior/behavior_metadata.py

```diff
@@ -139,3 +147,220 @@ def get_task_parameters(data: Dict) -> Dict:
+
+        :rtype: datetime
+        """
+        extractor_acq_date = self._extractor.get_date_of_acquisition()
+
+        pkl_data = self._behavior_stimulus_file
+        pkl_raw_acq_date = pkl_data["start_time"]
+        if isinstance(pkl_raw_acq_date, datetime):
+            pkl_acq_date = pytz.utc.localize(pkl_raw_acq_date)
+
+        elif isinstance(pkl_raw_acq_date, (int, float)):
+            # We are dealing with an older pkl file where the acq time is
+            # stored as a Unix style timestamp string
+            parsed_pkl_acq_date = datetime.fromtimestamp(pkl_raw_acq_date)
+            pkl_acq_date = pytz.utc.localize(parsed_pkl_acq_date)
+        else:
+            pkl_acq_date = None
+            warnings.warn(
+                "Could not parse the acquisition datetime "
+                f"({pkl_raw_acq_date}) found in the following stimulus *.pkl: "
+                f"{self._extractor.get_behavior_stimulus_file()}"
+            )
+
+        if pkl_acq_date:
+            acq_start_diff = (
+                    extractor_acq_date - pkl_acq_date).total_seconds()
+            # If acquisition dates differ by more than an hour
+            if abs(acq_start_diff) > 360:
+                session_id = self._extractor.get_behavior_session_id()
+                warnings.warn(
+                    "The `date_of_acquisition` field in LIMS "
+                    f"({extractor_acq_date}) for behavior session "
+                    f"({session_id}) deviates by more "
+                    f"than an hour from the `start_time` ({pkl_acq_date}) "
+                    "specified in the associated stimulus *.pkl file: "
+                    f"{self._extractor.get_behavior_stimulus_file()}"
+                )
+        return extractor_acq_date
+
+    @property
+    def reporter_line(self) -> Optional[str]:
+        """There can be multiple reporter lines, so it is returned from LIMS
+        as a list. But there shouldn't be more than 1 for behavior. This
+        tries to convert to str
+
+        Returns
+        ---------
+        single reporter line, or None if not possible
+        """
+        reporter_line = self._extractor.get_reporter_line()
+
+        if isinstance(reporter_line, str):
+            return reporter_line
+
+        if len(reporter_line) == 0:
+            warnings.warn('No reporter line')
+            return
+
+        if len(reporter_line) > 1:
+            warnings.warn('More than 1 reporter line. Returning the first one')
+
+        return reporter_line[0]
+
+    @property
+    def cre_line(self) -> Optional[str]:
+        """Parses cre_line from a list of driver_lines
+
+        Returns
+        ---------
+        cre_line, or None on error
+        """
+        if len(self.driver_line) == 0:
+            warnings.warn('Could not parse cre_line because there are no '
+                          'driver lines')
+            return
+        cre_line = [d for d in self.driver_line if d.endswith('Cre')]
```

Comment:
Is it possible for driver lines to represent `Cre` in different ways? Are string like `CRE` or `cre` valid?


Is it always the case that a `Cre` driver string will end with `Cre`? Is it possible for the `Cre` substring to be the start of a string or end up in the middle?

Reply:
For this release, they are definitely at the end.
Slc17a7-IRES2-Cre
Vip-IRES-Cre
Sst-IRES-Cre

And LIMS definitely has examples that are in the middle.  e.g. Emx1-CreERT2

I think it should probably be a `contains` instead of `endswith`

Reply:
@wbwakeman I found this existing code [here](https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/brain_observatory/behavior/swdb/behavior_project_cache.py#L144) which is part of behavior project cache, to parse cre line. It uses full_genotype. Is it reliable to use full_genotype to parse cre_line? I'd prefer not to have code duplication if possible.

Reply:
@aamster yes, good find.  Let's reuse that.  donor.full_genotype is a programmatically-generated string based on the genotype information in LIMS that is set by the lab people who do the actual genotyping experiment on the mouse (and set it in a different system called LabTracks)